File: 백준/Silver/1931. 회의실 배정/회의실 배정.java
import java.io.*;
import java.util.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
        StringTokenizer st;
        
        int N = Integer.parseInt(br.readLine());
        int[][] time = new int[N][2];
        
        for(int i = 0; i<N; i++) {
            st = new StringTokenizer(br.readLine(), " ");
            time[i][0] = Integer.parseInt(st.nextToken());
            time[i][1] = Integer.parseInt(st.nextToken());
        }
        
        Arrays.sort(time, new Comparator<int[]>() {
            @Override
            public int compare(int[] a, int[] b) {
                if(a[1] == b[1]) {
                    return a[0] - b[0];
                }
                return a[1] - b[1];
            }
        });
        
        int end_time = 0;
        int cnt = 0;
        for(int i=0; i<N; i++) {
            if(end_time <= time[i][0]) {
                end_time = time[i][1];
                cnt++;
            }
        }
        System.out.println(cnt);
    }
}

File: 백준/Silver/1931. 회의실 배정/회의실 배정.py
import sys

N = int(sys.stdin.readline())
meetings = []

for _ in range(N):
    a, b = map(int, sys.stdin.readline().split())
    meetings.append((a, b))

meetings.sort(key=lambda x: (x[1], x[0]))

cnt = 1
endT = meetings[0][1]

for i in range(1, N):
    if meetings[i][0] >= endT:
        endT = meetings[i][1]
        cnt += 1
        
print(cnt)